app/controllers/katello/api/v2/host_module_streams_controller.rb

```diff
@@ -22,6 +22,9 @@ def index_relation
       return HostAvailableModuleStream.upgradable([@host]) if params[:status] == HostAvailableModuleStream::UPGRADABLE
 
       rel = @host.host_available_module_streams
+      if params[:sort_by] == 'installed_profiles'
+        rel = rel.order([:installed_profiles, :status])
```

Comment:
Ordering by two fields produces a sort order close enough to what we want. But the scoped_search `params[:order_by]` doesn't accept multiple fields for ordering. So I did the sorting in `index_relation` here instead. Seems to work well.